plugins/luxtronic2/__init__.py

```diff
@@ -226,51 +247,101 @@ def refresh_calculated(self):
 
 
 class Luxtronic2(LuxBase):
-    _parameter = {}
-    _attribute = {}
-    _calculated = {}
+    _parameter = defaultdict(lambda: defaultdict(int))
+    _attribute = defaultdict(lambda: defaultdict(int))
+    _calculated = defaultdict(lambda: defaultdict())
     _decoded = {}
     alive = True
 
-    def __init__(self, smarthome, host, port=8888, cycle=300):
-        LuxBase.__init__(self, host, port)
+    def __init__(self, smarthome, host, port=8888,
+                 cycle=300, reconnect_cycle=60, max_timeouts=10):
+        LuxBase.__init__(self, host, port, max_timeouts)
         self._sh = smarthome
         self._cycle = int(cycle)
-        self.connect()
+        self._reconnect_cycle = int(reconnect_cycle)
+        try:
+            self.connect()
+        except Exception as e:
+            logger.error('luctronic2 connect faild: {}'.format(e))
 
     def run(self):
         self.alive = True
-        self._sh.scheduler.add('Luxtronic2', self._refresh, cycle=self._cycle)
+        # Letting the scheduler recall the foo by handing it "cycle" could
+        # fail in this application, because it calles the foo without checking
+        # if the last call finished. This could clog up the system.
+        threading.Thread(name='Lux2_cycle', target=self.__refresh).start()
+        threading.Thread(name='Lux2_recon', target=self._reconnect).start()
```

Comment:
See also my [bug report](https://github.com/mknx/smarthome/issues/208) for that. For now, it works with threading, but I could not get it working with the scheduler. I don't think it is about the socket timeout, because I also had problems with the scheduler in the modbus plugin too.